src/constants.h

```diff
@@ -55,11 +55,14 @@ constexpr char kOpenVINORuntimeBackend[] = "openvino";
 
 constexpr char kPyTorchLibTorchPlatform[] = "pytorch_libtorch";
 constexpr char kPyTorchLibTorchFilename[] = "model.pt";
+constexpr char kPyTorchPythonPlatform[] = "pytorch_python";
```

Comment:
Is this just for symmetry? I thought the `runtime` flag essentially covered what behavior to follow. Are we also introducing this platform flag?

Reply:
This is a good point. After autocomplete, the [model config is validated](https://github.com/triton-inference-server/core/pull/285/files#diff-f947d03afcf01249667bedc5b955404da083564c22809b6666ba923356f7c829L1310), which [takes the platform value and compare it againsts the backend value](https://github.com/triton-inference-server/core/pull/285/files#diff-f947d03afcf01249667bedc5b955404da083564c22809b6666ba923356f7c829R1425). If the runtime is Python, it would be strange to use the LibTorch platform, but [keeping it empty will fail the check](https://github.com/triton-inference-server/core/pull/285/files#diff-f947d03afcf01249667bedc5b955404da083564c22809b6666ba923356f7c829L2263), so it is there to keep the validation logic happy.

Reply:
Does this mean if we wanted to add tensorflow python runtime implementation, we'd also need to introduce `platform: "tensorflow_python"` for it to work? Is there a clean way to not require the additional platform flag for these?

I think it also checks for backend field right?

I believe expectation is that users will specify:
```
backend: "pytorch"
```
and there is backwards compatibility to support:
```
platform: "pytorch_libtorch" # --> backend: pytorch
```

If both are left unspecified, we would check file extension somewhere like `.pt` or something to infer pytorch backend. Then if all else fails, we cannot infer the backend and they need to specify it in the config.

I don't think we need to add `platform: pytorch_python` to satisfy this flow, do we?

Reply:
I removed the "pytorch_python" platform, and added a comment above saying new runtimes should ignore this field, which defaults for "pytorch_libtorch", as the platform may not be LibTorch.